### PR TITLE
planner, executor: indexMerge support partition table with limit embedded

### LINKED ISSUE #43158: Wrong result for partition table with order by + limit
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create table t(a int, b int, index idx_bc(a, b)) PARTITION BY HASH (`b`) PARTITIONS 4;
insert into t values (1,2),(1,3),(1,1),(1,5);
analyze table t;
select * from t where a = 1 order by b limit 10;
```

### 2. What did you expect to see? (Required)

```
mysql> select * from tcommonhash where a = 1 order by b limit 10;
+------+------+
| a    | b    |
+------+------+
|    1 |    1 |
|    1 |    2 |
|    1 |    3 |
|    1 |    5 |
+------+------+
4 rows in set (0.00 sec)
```

### 3. What did you see instead (Required)

```
mysql> select * from tcommonhash where a = 1 order by b limit 10;
+------+------+
| a    | b    |
+------+------+
|    1 |    1 |
|    1 |    3 |
|    1 |    5 |
|    1 |    2 |
+------+------+
4 rows in set (0.00 sec)
```

### 4. What is your TiDB version? (Required)

Only on master branch

### LINKED ISSUE #43160: TiKV reports error when executor IndexMerge with order by + limit
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Use tikv to execute SQLs below 
```
create table tcommon(a int, b int, c int, d int auto_increment, primary key(a, c, d), index idx_ac(a, c), index idx_bc(b, c));
insert into tcommon(a,b,c) values (1,1,1),(2,2,2),(1,1,2),(2,2,1),(1,1,5);
analyze table tcommon;
select * from tcommon where a = 1 or b = 2 order by c limit 3;
```

### 2. What did you expect to see? (Required)

no error

### 3. What did you see instead (Required)
get error msg below
```
ERROR 1105 (HY000): [components/tidb_query_executors/src/table_scan_executor.rs:425]: Data is corrupted, missing data for NOT NULL column (offset = 0)
```
### 4. What is your TiDB version? (Required)

only on master branch

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #43158, close #43160, ref #41028

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```
use real tikv env to run executor_test.TestOrderByWithLimit and passed.
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: executor/index_merge_reader.go [hunk 1]
```diff
@@ -593,15 +604,22 @@ func (e *IndexMergeReaderExecutor) getTablePlanRootID() int {
...
+	tableReader        Executor
+	memTracker         *memory.Tracker
+	partitionTableMode bool
+	prunedPartitions   []table.PhysicalTable
+	byItems            []*plannerutil.ByItems
+	scannedKeys        uint64
+	pushedLimit        *plannercore.PushedDownLimit
+}
+
+func (w *partialTableWorker) needPartitionHandle() bool {
```
Comment (XuHuaiyu) [inline executor/index_merge_reader.go:621]: add a comment for this function.

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**